#VRML_SIM R2025a utf8

EXTERNPROTO "protos/Backgrounds/TexturedBackground.proto"
EXTERNPROTO "protos/Backgrounds/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/robotstadium/protos/RobotstadiumSoccerField.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/balls/protos/RobocupSoccerBall.proto"
EXTERNPROTO "protos/Nao/Nao.proto"
EXTERNPROTO "protos/Human/CharacterSkin.proto"

WorldInfo {
  info [
    "Simulation of the Robocup Standard Platform League"
  ]
  title "Robocup"
  basicTimeStep 20
  contactProperties [
    ContactProperties {
      material1 "NAO foot material"
      coulombFriction [
        7
      ]
      bounce 0.3
      bounceVelocity 0.003
    }
  ]
}
Viewpoint {
  orientation -0.004985986175514832 -0.9971889251142134 0.07476220683886588 6.149625664378425
  position -0.6921095650273519 -0.024903451109388075 1.328651458248591
  follow "soccer ball"
}
TexturedBackground {
}
DEF TEXTURED_BACKGROUND_LIGHT TexturedBackgroundLight {
  luminosity 1.5
  direction -1 -1 1
}
RobotstadiumSoccerField {
  rotation 0 0 1 1.5707963267948966
  frame1Color 0.9 0.8 0.2
  frame2Color 0.2 0.4 0.8
}
Solid {
  translation 3.3 4 1.25
  rotation 0 -1 0 1.5708
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0.8
        baseColorMap ImageTexture {
          url [
            "./referee_background/mediumlight_uncrowded_left_0.png"
          ]
          filtering 5
        }
        roughness 1
        metalness 0
        textureTransform TextureTransform {
          rotation 1.5708
        }
      }
      geometry Plane {
        size 3 4
      }
    }
  ]
  name "solid(1)"
}
Solid {
  translation 3.3 0 1
  rotation 0 -1 0 1.5708
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0.8
        baseColorMap ImageTexture {
          url [
            "./referee_background/mediumlight_uncrowded_middle_2.png"
          ]
          filtering 5
        }
        roughness 1
        metalness 0
        textureTransform TextureTransform {
          rotation 1.5708
        }
      }
      geometry Plane {
        size 3 4
      }
    }
  ]
  name "solid(2)"
}
Solid {
  translation 3.3 -4 1
  rotation 0 -1 0 1.5708
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0.8
        baseColorMap ImageTexture {
          url [
            "./referee_background/mediumlight_uncrowded_right_0.png"
          ]
          filtering 5
        }
        roughness 1
        metalness 0
        textureTransform TextureTransform {
          rotation 1.5708
        }
      }
      geometry Plane {
        size 3 4
      }
    }
  ]
  name "solid(3)"
}
DEF PLAYER_RED_2 Nao {
  translation 0.7 1.14 0.3
  rotation 0 0 -1 0.56
  name "NAO RED 2"
  customColor [
    1 0 0
  ]
  controller "nao_soccer_player"
  supervisor TRUE
}
DEF PLAYER_RED_3 Nao {
  translation 0.00941066 -0.00755316 0.305915
  rotation -0.01877604437981655 0.05548191248081572 0.9982831349596759 0.04299641178735337
  name "NAO RED 3"
  customColor [
    1 0 0
  ]
  controller "nao_soccer_player"
  supervisor TRUE
}
DEF PLAYER_BLUE_4 Nao {
  translation 0.7 -1.14 0.3
  rotation 0 0 1 0.56
  name "NAO BLUE 4"
  customColor [
    0 0 1
  ]
  controller "nao_soccer_player"
  supervisor TRUE
}
DEF OBSTACLE_ROBOT Nao {
  translation 0 1.5 0.3
  rotation 0 0 1 0.56
  name "OBSTACLE ROBOT"
  customColor [
    0 0 0
  ]
  controller "nao_soccer_player"
  supervisor TRUE
}
DEF ANTHONY1 Robot {
  translation 3 0 0
  rotation 0 0 1 3.14159
  children [
    CharacterSkin {
      name "Anthony"
      model "Anthony"
    }
  ]
  name "anthony"
  controller "bvh_animation"
  controllerArgs [
    "-d"
    "Anthony"
    "-f"
    "../../motions/corner_kick_red.bvh"
    "-l"
  ]
  supervisor TRUE
}
DEF SOCCER_BALL RobocupSoccerBall {
  translation -0.302267 2.11301 0.0697989
  rotation 0.6890254618190954 0.22469496195991764 0.6890254618189474 2.6995443309089437
}
